filament/backend/src/vulkan/VulkanDriver.cpp

```diff
@@ -989,9 +994,38 @@ void VulkanDriver::endRenderPass(int) {
     assert(mCurrentRenderTarget);
     vkCmdEndRenderPass(mContext.currentCommands->cmdbuffer);
     mCurrentRenderTarget = VK_NULL_HANDLE;
+    if (mContext.currentRenderPass.currentSubpass > 0) {
+        for (uint32_t i = 0; i < VulkanBinder::TARGET_BINDING_COUNT; i++) {
+            mBinder.bindInputAttachment(i, {});
+        }
+        mContext.currentRenderPass.currentSubpass = 0;
+    }
     mContext.currentRenderPass.renderPass = VK_NULL_HANDLE;
 }
 
+void VulkanDriver::nextSubpass(int) {
+    assert(mContext.currentCommands);
+    assert(mContext.currentSurface);
+    assert(mCurrentRenderTarget);
+    assert(mContext.currentRenderPass.subpassMask);
```

Comment:
should we also assert() mContext.currentRenderPass.currentSubpass ? It's limited to 1 right now, right?

Reply:
Yes, currently I would expect `currentSubpass` to always be zero, but in the future we might allow several subpasses.

Reply:
I'm just saying if we do several calls to nextSubpass() in filament (by accident), it would be good to be caught here (and also make sure that it doesn't cause out-of-bounds accesses (even in release)).

Reply:
Understood, amended the commit -- since you mentioned release, I used `ASSERT_PRECONDITION` for this instead of `assert`.